### PR TITLE
DOC: Cleaned docstrings in core/indexers.py

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/indexers.py [hunk 1]
```diff
@@ -66,28 +70,25 @@ def check_setitem_lengths(indexer, value, values) -> None:
     """
     Validate that value and indexer are the same length.
 
-    An special-case is allowed for when the indexer is a boolean array
-    and the number of true values equals the length of ``value``. In
-    this case, no exception is raised.
+    A special-case is allowed for when the indexer is a
```
Comment (WillAyd) [inline pandas/core/indexers.py]: I think this section in particular read better with the previous text, so OK to revert

File: pandas/core/indexers.py [hunk 2]
```diff
@@ -66,28 +70,25 @@ def check_setitem_lengths(indexer, value, values) -> None:
...
-    -------
-    None
+        Values being set into.
 
     Raises
     ------
     ValueError
-        When the indexer is an ndarray or list and the lengths don't
-        match.
+        When the indexer is a ndarray or list and the
```
Comment (WillAyd) [inline pandas/core/indexers.py]: Can revert this back as well

File: pandas/core/indexers.py [hunk 3]
```diff
@@ -168,27 +168,30 @@ def maybe_convert_indices(indices, n: int):
...
-    IndexError : one of the converted indices either exceeded the number
-        of elements (specified by `n`) OR was still negative.
-    """
+    IndexError
+        One of the converted indices either exceeded the number of,
+        elements (specified by `n`), or was still negative.
 
+    Notes
+    -----
+    If `indices` is empty, `np.array` will return a float,
```
Comment (WillAyd) [inline pandas/core/indexers.py]: I think this was better placed as a code comment before

**Review by WillAyd — CHANGES_REQUESTED** [review]
Looks good - a few minor things I think should be reverted

File: pandas/core/indexers.py [hunk 4]
```diff
@@ -65,29 +69,24 @@ def is_empty_indexer(indexer, arr_value: np.ndarray) -> bool:
 def check_setitem_lengths(indexer, value, values) -> None:
     """
     Validate that value and indexer are the same length.
-
-    An special-case is allowed for when the indexer is a boolean array
-    and the number of true values equals the length of ``value``. In
-    this case, no exception is raised.
+    A special-case is allowed for when the indexer is a
+    boolean array, and the number of true values, equals to
+    the length of ``value``.
```
Comment (jorisvandenbossche) [inline pandas/core/indexers.py]: The previous sentence is still correcter english I think (or at least the comma after "values" is not in a correct place)

File: pandas/core/indexers.py [hunk 5]
```diff
@@ -65,29 +69,24 @@ def is_empty_indexer(indexer, arr_value: np.ndarray) -> bool:
...
-    -------
-    None
+        Values being set into.
 
     Raises
     ------
     ValueError
-        When the indexer is an ndarray or list and the lengths don't
-        match.
+        When the indexer is a ndarray or list and the lengths don't match.
```
Comment (jorisvandenbossche) [inline pandas/core/indexers.py]:
```suggestion
        When the indexer is an ndarray or list and the lengths don't match.
```

"an" is actually correct here I think

Comment (MomIsBestFriend) [reply pandas/core/indexers.py]: @jorisvandenbossche English is not my native language. All I know that's you put ```a``` before words that starts with ```A,E,I,O,U``` (probably missed a letter or two).
If it's such a big of a deal, I don't mind reverting, besides you shouldn't take my work for it, my grammar isn't great.

File: pandas/core/indexers.py (hunk 4 above)
Comment (MomIsBestFriend) [reply pandas/core/indexers.py]: @jorisvandenbossche I'm sorry, I don't understand what should I fix here? can you be more specific?

File: pandas/core/indexers.py (hunk 5 above)
Comment (jorisvandenbossche) [reply pandas/core/indexers.py]: I am also no native english speaker, but I know that the a vs an depends on the pronounciation, not spelling (see eg https://writingexplained.org/a-vs-an-difference). 
And in this case, I actually looked at numpy's docs, and they clearly use "an ndarray" much more than "a ndarray".

File: pandas/core/indexers.py (hunk 4 above)
Comment (jorisvandenbossche) [reply pandas/core/indexers.py]: There is still a diff show here, and the original version is a correcter sentence I think. The comma (",") in "the number of true values, equals .." is not correct

File: pandas/core/indexers.py (hunk 5 above)
Comment (MomIsBestFriend) [reply pandas/core/indexers.py]: Thank you so much for the patience!

Do you mind if I write a test case for that?

File: pandas/core/indexers.py (hunk 4 above)
Comment (MomIsBestFriend) [reply pandas/core/indexers.py]: I see it now, you're 100% correct.

File: pandas/core/indexers.py [hunk 6]
```diff
@@ -65,29 +69,24 @@ def is_empty_indexer(indexer, arr_value: np.ndarray) -> bool:
 def check_setitem_lengths(indexer, value, values) -> None:
     """
     Validate that value and indexer are the same length.
-
-    An special-case is allowed for when the indexer is a boolean array
-    and the number of true values equals the length of ``value``. In
-    this case, no exception is raised.
+    A special-case is allowed for when the indexer is a
+    boolean array, and the number of true values equals to
```
Comment (jorisvandenbossche) [inline pandas/core/indexers.py]:
```suggestion
    boolean array and the number of true values equals
```

File: pandas/core/indexers.py [hunk 7]
```diff
@@ -208,7 +206,11 @@ def maybe_convert_indices(indices, n: int):
 
 def length_of_indexer(indexer, target=None) -> int:
     """
-    return the length of a single non-tuple indexer which could be a slice
+    Returns
```
Comment (jorisvandenbossche) [inline pandas/core/indexers.py]: Docstrings ideally start with a one liner that says what the function does. Can you leave something like that?

**Review by jorisvandenbossche — CHANGES_REQUESTED** [review]

File: pandas/core/indexers.py (hunk 1 above)
Comment (WillAyd) [reply pandas/core/indexers.py]: I think should revert altogether. not explicitly checked here but our docstring validator requires a blank line between short summary and extended description, which this does away with

The old formatting was fine so can just leave as is

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/core/indexers.py [hunk 8]
```diff
@@ -65,29 +69,24 @@ def is_empty_indexer(indexer, arr_value: np.ndarray) -> bool:
 def check_setitem_lengths(indexer, value, values) -> None:
     """
     Validate that value and indexer are the same length.
-
```
Comment (simonjayhawkins) [inline pandas/core/indexers.py]: can you revert this line removal. Multi-line docstrings consist of a summary line just like a one-line docstring, followed by a blank line, followed by a more elaborate description. see https://www.python.org/dev/peps/pep-0257/

File: pandas/core/indexers.py [hunk 9]
```diff
@@ -27,8 +27,12 @@ def is_list_like_indexer(key) -> bool:
 
 
 def is_scalar_indexer(indexer, arr_value) -> bool:
-    # return True if we are all scalar indexers
-
+    """
+    Returns
+    -------
+    bool
+        True if X are all scalar indexers.
```
Comment (simonjayhawkins) [inline pandas/core/indexers.py]: what is X?

File: pandas/core/indexers.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/core/indexers.py]: This is still showing a diff.

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]

**Comment by jbrockmendel** [comment]
@MomIsBestFriend can you rebase

File: pandas/core/indexers.py (hunk 1 above)
Comment (MomIsBestFriend) [reply pandas/core/indexers.py]: > This is still showing a diff.

Just because I changed the ```An``` to ```A``` at the very first line

**Comment by MomIsBestFriend** [comment]
rebased @jbrockmendel

**Comment by MomIsBestFriend** [comment]
Thank you @jorisvandenbossche

**Comment by jorisvandenbossche** [comment]
(Will merge when green)

**Comment by jorisvandenbossche** [comment]
Thanks @MomIsBestFriend !

**Pull request merged**